TypeSystem/flow-analysis/reachability_conditional_A04_t01.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Conditional statement: If `N` is a conditional statement of the
+/// form `if (E) S1 else S2` then:
+/// - Let `before(E) = before(N)`.
+/// - Let `before(S1) = split(true(E))`.
+/// - Let `before(S2) = split(false(E))`.
+/// - Let `after(N) = merge(after(S1), after(S2))`.
+///
+/// @description Checks that `before(S1) = split(true(E))` and
+/// `before(S2) = split(false(E))`.
+/// @author sgrekhov22@gmail.com
+
+Never foo() => throw "Never";
+
+main() {
+  int i;
+  bool b = true;
+  print(b); // no assignment to `b`
+  if (b) {
+    i = 42;
+  } else {
+    foo();
+  }
+  i; // Definitely assigned
```

Comment:
Wouldn't this test be slightly stronger if the flow analysis does not "know" that `b` has the value `true`?

Yes, again, it shouldn't rely on this knowledge, but if the flow analysis (wrongly) relies on this knowledge then we'd still conclude that `i` is definitely assigned below.

If we could instead have a situation where the flow analysis surely does not know the value of `b` then the test that `i` is definitely assigned would be known to take the return type `Never` of `foo` into account.

I think it would be enough to use `bool b = 2 > 1;`. Another way could be to make `b` a top-level variable.

Reply:
Ah, yes! Of course! Thank you. Updated.